worker/dbaccessor/worker.go

```diff
@@ -831,14 +837,64 @@ func (w *dbWorker) ensureNamespace(namespace string) error {
 	return nil
 }
 
+func (w *dbWorker) ensureModelDBInitialised(namespace string) error {
+	db, err := w.txnRunnerForNamespace(namespace)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	ctx, cancel := w.scopedContext()
+	defer cancel()
+
+	// Check if the DB has metadata for one of our known tables.
+	var runDBMigration bool
+	if err := errors.Trace(db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		row := tx.QueryRowContext(ctx, "SELECT 1 FROM sqlite_master WHERE name='change_log'")
```

Comment:
I really think we'll need to run patch migrations against a schema table. For now, this is more than enough. In fact I'm doing this same thing in the pruner.